### PR TITLE
Implement a simple schema migration system for PostgreSQL (fixes #139)

### LINKED ISSUE #139: Basic schema migration implementation in PG storage backend
I propose that we store a ``schema_version`` (simple integer) in the ``metadata`` table.

Then on startup, instead of checking the existance of the ``records`` like we do currently, we compare the schema versions.

For each missing intermediary version, we run the script ``migration_N_N+1.sql``. Each migration script would update the ``schema_version`` value.

The ``schema.sql`` would always contain the last version of the schema and end up by setting the ``schema_version`` value.

### PR DESCRIPTION
**Pull request by leplatrem**

### CODE REVIEW

**Comment by ametaireau** [comment]
And of course I commented on the wrong one :p

**Comment by leplatrem** [comment]
@Natim @ametaireau r?

**Comment by Natim** [comment]
LGTM

File: cliquet/scripts/cliquet.py [hunk 1]
```diff
@@ -2,10 +2,17 @@
 import argparse
 import sys
 import textwrap
+import warnings
 
 from pyramid.paster import bootstrap
 
 
+def deprecated_init(env):
+    message = '"cliquet init" is deprecated. Use "cliquet migrate" instead.'
```
Comment (ametaireau) [inline cliquet/scripts/cliquet.py]: migrate sounds weird for an initializer. I would have expected to have two commands: init and migrate (not related)

File: cliquet/storage/postgresql/__init__.py [hunk 2]
```diff
@@ -135,27 +137,43 @@ def __init__(self, *args, **kwargs):
 
     def _execute_sql_file(self, filepath):
         here = os.path.abspath(os.path.dirname(__file__))
-        schema = open(os.path.join(here, 'schema.sql')).read()
+        schema = open(os.path.join(here, filepath)).read()
```
Comment (ametaireau) [inline cliquet/storage/postgresql/__init__.py]: good catch!

**Comment by ametaireau** [comment]
Neat.

Having a migration script is indeed something we want. I'm not sure about the rename of `init` in `migrate`.
What do you think about having two commands `cliquet initdb` and `cliquet migratedb`, the former could accept an optional ``--migrate`` option.

File: cliquet/scripts/cliquet.py (hunk 1 above)
Comment (leplatrem) [inline cliquet/scripts/cliquet.py]: That's what Django has. ``manage.py migrate`` for everything :)

Do you suggest to separate the init process from the migration too ?

**Comment by ametaireau** [comment]
I'll let you rebase and then you can merge :)

**Pull request merged**